controllers/productController.js

```diff
@@ -0,0 +1,58 @@
+import { EventEmitter } from 'events';
+import path from 'path';
+import fileHelper from '../helpers/fileHelper';
+
+const PATH_TO_PRODUCTS = path.resolve(__dirname, '../models/products.json');
+export const EVENTS = {
+    getProducts: 'getProducts',
+    setProduct: 'setProduct',
+    getProductId: 'getProductId',
+    reviews: 'reviews',
+    read: 'read',
+    write: 'write',
+}
+
+class ProductController extends EventEmitter {
+
+    getProducts() {
+        fileHelper.readFromFile(PATH_TO_PRODUCTS, this, EVENTS.getProducts);
+    }
+
+    getProductId(id) {
+        fileHelper.readFromFile(PATH_TO_PRODUCTS, this, EVENTS.read);
```

Comment:
Well, it's super strange approach with events. What if there will be two parallel requests - one for product, one for review and order of `EVENTS.read` will be different?